indexing-service/src/main/java/org/apache/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -508,13 +508,14 @@ public int getNumSplits()
     // segments to olders.
 
     // timelineSegments are sorted in order of interval
-    int index = 0;
+    int[] index = {0};
     for (TimelineObjectHolder<String, DataSegment> timelineHolder : Lists.reverse(timelineSegments)) {
       for (PartitionChunk<DataSegment> chunk : timelineHolder.getObject()) {
         for (String metric : chunk.getObject().getMetrics()) {
-          if (!uniqueMetrics.containsKey(metric)) {
-            uniqueMetrics.put(metric, index++);
+          uniqueMetrics.computeIfAbsent(metric, k -> {
+            return index[0]++;
           }
```

Comment:
Using index variable as int as is, the compiler complains `"Variables used in lambda should be final or effectively final"`. The fix is to use an integer array with one element. Let me know if this is right.